### ISSUE TITLE
Unhandled exception (#0da538f1)

### ISSUE THREAD

**Issue by sqlmapreporter** [sqlmapproject/sqlmap#3197]
```
Running version: 1.2.3.27#dev
Python version: 2.7.14
Operating system: Windows-2012ServerR2-6.3.9600
Command line: sqlmap.py --url=[REDACTED] --crawl=1 --threads=10 --retries=0 --timeout=5 --risk=3 --level=5 --dbms=MYSQL --technique=EU --random-agent --is-dba --exclude-sysdbs --answers=remove=Y,follow=N,optimize=Y,want=Y,provided=1,which=a,thread=10,dictionary-based=n,problems=Y,problem=N,redirect=n,do you want to exploit this SQL injection=Y,do you want to test this URL=Y,retrieved=n,WAF=N,check=n,include=n,xml=y,do you want to check for the existence of site's sitemap=y,injection not exploitable with NULL values=n,store crawling results=n,do you want to test URL=Y,how do you want to proceed=C,you want to keep testing the others=y,you want to skip test payloads sp ecific for other DBMSes=Y,do you want to include all tests for=n,do you want sqlmap to try to optimize value(s) for DBMS delay responses=n,do you want to dump entries=Y,which database(s)=a,which table(s) of database=a
Technique: UNION
Back-end DBMS: MySQL (identified)
```
```
Traceback (most recent call last):
  File "sqlmap.py", line 157, in main
    start()
  File "lib/controller/controller.py", line 548, in start
    injection = checkSqlInjection(place, parameter, value)
  File "lib/controller/checks.py", line 670, in checkSqlInjection
    reqPayload, vector = unionTest(comment, place, parameter, value, prefix, suffix)
  File "lib/techniques/union/test.py", line 314, in unionTest
    validPayload, vector = _unionTestByCharBruteforce(comment, place, parameter, value, prefix, suffix)
  File "lib/techniques/union/test.py", line 273, in _unionTestByCharBruteforce
    count = _findUnionCharCount(comment, place, parameter, value, prefix, suffix, PAYLOAD.WHERE.ORIGINAL if isNullValue(kb.uChar) else PAYLOAD.WHERE.NEGATIVE)
  File "lib/techniques/union/test.py", line 108, in _findUnionCharCount
    page, headers, code = Request.queryPage(payload, place=place, content=True, raise404=False)
  File "lib/request/connect.py", line 1233, in queryPage
    page, headers, code = Connect.getPage(url=uri, get=get, post=post, method=method, cookie=cookie, ua=ua, referer=referer, host=host, silent=silent, auxHeaders=auxHeaders, response=response, raise404=raise404, ignoreTimeout=timeBasedCompare)
  File "lib/request/connect.py", line 467, in getPage
    conn = urllib2.urlopen(req)
  File "Python27/lib/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "Python27/lib/urllib2.py", line 435, in open
    response = meth(req, response)
  File "Python27/lib/urllib2.py", line 548, in http_response
    'http', request, response, code, msg, hdrs)
  File "Python27/lib/urllib2.py", line 467, in error
    result = self._call_chain(*args)
  File "Python27/lib/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "lib/request/redirecthandler.py", line 91, in http_error_302
    content = decodePage(content, headers.get(HTTP_HEADER.CONTENT_ENCODING), headers.get(HTTP_HEADER.CONTENT_TYPE))
  File "lib/request/basic.py", line 307, in decodePage
    metaCharset = checkCharEncoding(extractRegexResult(META_CHARSET_REGEX, page))
  File "lib/core/decorators.py", line 20, in _
    cache[key] = f(*args, **kwargs)
  File "lib/request/basic.py", line 161, in checkCharEncoding
    encoding = encoding.lower()
AttributeError: 'float' object has no attribute 'lower'
```

**Comment by AmericanY** [comment]
@stamparm looks like you need to edit `--url` with `sqlmapreporter` to replace the target with asterisks `*` upon reporting it. as i see the user have no knowledge of using `sqlmap`.

**Comment by stamparm** [comment]
The issue itself should be patched with 4147f44e6359ac0ecd42a4aea79a4016d5617eee (1.2.4.1)
